src/pages/Vote/VotingWallet/index.tsx

```diff
@@ -64,20 +133,63 @@ export const VoteUi: React.FC<IVoteUiProps> = ({ votingWeight, xvsLockedWei }) =
           <Trans
             i18nKey="vote.delegateYourVoting"
             components={{
-              Anchor: <span css={styles.clickableText} role="button" aria-pressed="false" />,
+              Anchor: (
+                <span
+                  css={styles.clickableText}
+                  role="button"
+                  aria-pressed="false"
+                  tabIndex={0}
+                  onClick={() => setDelegateModelIsOpen(true)}
+                />
+              ),
             }}
           />
         </Typography>
       </Paper>
+      <DelegateModal
+        onClose={() => setDelegateModelIsOpen(false)}
+        isOpen={delegateModelIsOpen}
+        currentUserAccountAddress={currentUserAccountAddress}
+        previouslyDelegated={previouslyDelegated}
+        setVoteDelegation={setVoteDelegation}
+      />
     </div>
   );
 };
 
-const Vote: React.FC = () => (
-  <VoteUi
-    votingWeight={new BigNumber(1.003)}
-    xvsLockedWei={new BigNumber('19931200345567000000')}
-  />
-);
+const VotingWallet: React.FC = () => {
+  const { account: { address: accountAddress } = { address: undefined }, openAuthModal } =
+    useContext(AuthContext);
+
+  const { data: currentVotesWei } = useGetCurrentVotes(
+    { accountAddress: accountAddress || '' },
+    { enabled: !!accountAddress },
+  );
+  const { data: delegate } = useGetVoteDelegates(
+    { accountAddress: accountAddress || '' },
+    { enabled: !!accountAddress },
+  );
+
+  const { data: vaults } = useGetVestingVaults({ accountAddress });
+  const [xvsVault] = vaults.filter(v => v.stakedTokenId === XVS_TOKEN_ID);
+
+  const { mutateAsync: setVoteDelegation } = useSetVoteDelegate({
+    onSettled: () => {},
```

Comment:
Can be removed

Reply:
ahh yeah this was supposed to remind me to take care of your following comment haha